### PR TITLE
[PLUGIN-632] Removed the retrying logic from config and improved plugin documentation

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-632](https://cdap.atlassian.net/browse/PLUGIN-632)

### CODE REVIEW

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java [hunk 1]
```diff
@@ -111,6 +111,6 @@ public void checkRootFolder() throws IOException {
   }
 
   public void isFolderAccessible(String folderId) throws IOException {
-    service.files().get(folderId).execute();
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java]: why is this change needed?

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveFilteringClient.java [hunk 2]
```diff
@@ -64,8 +64,8 @@ public GoogleDriveFilteringClient(C config) throws IOException {
    */
   public List<File> getFilesSummary(List<ExportedType> exportedTypes, int filesNumber)
       throws ExecutionException, RetryException {
-    Retryer<List<File>> filesSummaryRetryer = APIRequestRetryer.getRetryer(config,
-        String.format("Get files summary, files: '%d'.", filesNumber));
+    Retryer<List<File>> filesSummaryRetryer = APIRequestRetryer.getRetryer(String.format("Get files summary, files:" +
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/google/common/GoogleDriveFilteringClient.java]: nit: fix formatting

File: src/main/java/io/cdap/plugin/google/drive/source/GoogleDriveSourceClient.java [hunk 3]
```diff
@@ -76,9 +76,9 @@ public FileFromFolder getFile(String fileId) throws IOException, ExecutionExcept
    * @throws RetryException if there was an error getting the column information for the retry
    */
   public FileFromFolder getFilePartition(String fileId, Long bytesFrom, Long bytesTo)
-      throws IOException, ExecutionException, RetryException {
-    Retryer<FileFromFolder> fileFromFolderRetryer = APIRequestRetryer.getRetryer(config,
-        String.format("File retrieving, id: '%s'.", fileId));
+      throws ExecutionException, RetryException {
+    Retryer<FileFromFolder> fileFromFolderRetryer = APIRequestRetryer.getRetryer(String.format("File retrieving, " +
+                                                                                                 "id: '%s'.", fileId));
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/google/drive/source/GoogleDriveSourceClient.java]: nit: fix formatting

**Comment by rmstar** [comment]
Can you please add sample pipelines under examples directory?

File: docs/GoogleDrive-batchsink.md [hunk 4]
```diff
@@ -20,22 +20,24 @@ Is optional. In the case it is not set files have randomly generated 16-symbols
...
+URL. For example, if the URL was “https://drive.google.com/drive/folders/1dyUEebJaFnWa3Z4n0BFMVAXQ7mfUH11g”, then the 
+Directory Identifier would be “1dyUEebJaFnWa3Z4n0BFMVAXQ7mfUH11g”.
 
 ### Authentication
 
-**Authentication Type:** Type of authentication used to access Google API. 
+**Authentication Type:** Type of authentication used to access Google API. Make sure Google Drive Api is enabled.
 OAuth2 and Service account types are available.
 
 #### OAuth2 Properties
```
Comment (rmstar) [inline docs/GoogleDrive-batchsink.md:32]: Can you add some details on how to generate oauth token?

File: src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java (hunk 1 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/google/common/GoogleDriveClient.java]: Not needed, added while debugging, forgot to remove. Will remove that change in future commit.

File: docs/GoogleDrive-batchsink.md (hunk 4 above)
Comment (itsankit-google) [reply docs/GoogleDrive-batchsink.md:32]: Acknowledged

**Comment by itsankit-google** [comment]
> Can you please add sample pipelines under examples directory?

Acknowledged

File: examples/GoogleSheets-cdap-data-pipeline.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                    "type": "batchsource",
+                    "label": "Google Sheets Source",
+                    "artifact": {
+                        "name": "google-drive-plugins",
+                        "version": "1.4.0-SNAPSHOT",
+                        "scope": "USER"
+                    },
+                    "properties": {
+                        "referenceName": "GoogleSheetsSourceExample",
+                        "directoryIdentifier": "1Yis5kxpDNpanlTuXbJVBMpq9ltQXeCut",
```
Comment (rmstar) [inline examples/GoogleSheets-cdap-data-pipeline.json]: Let's leave this blank in the sample pipeline, user needs to set this. Same thing in the sample pipeline for google drive

Comment (itsankit-google) [reply examples/GoogleSheets-cdap-data-pipeline.json]: Acknowledged

File: src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java [hunk 6]
```diff
@@ -186,6 +189,12 @@ private boolean validateServiceAccount(FailureCollector collector) {
     final Boolean serviceAccountFilePath = isServiceAccountFilePath();
     final Boolean serviceAccountJson = isServiceAccountJson();
 
+    // we don't the validation to fail because the VM used during the validation
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java]: nit: don't the -> don't want the.

**Review by rmstar — APPROVED** [review]
Please squash commits after fixing minor comment

File: src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java (hunk 6 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java]: Acknowledged
